ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -2118,3 +2099,30 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
         self.assertEqual(suborder_data['Material Type'], material_type)
         self.assertEqual(suborder_data['Article Name'], article)
         self.assertEqual(suborder_data['Test Units'], test_unit)
+
+    def test035_archived_test_unit_shoudnt_display_in_the_order_drop_down_list(self):
+        """
+        Orders: Archived Test unit: Archive Approach: Archived test units shouldn't appear in orders in the drop down list
+        LIMS-3710
+        :return:
+        """
+        testunits, payload = self.test_unit_api.get_all_test_units(limit=20, deleted=1)
+        random_row = random.choice(testunits['testUnits'])
+
+        self.order_page.get_orders_page()
+        self.base_selenium.LOGGER.info('select random order record')
+        orders, payload = self.orders_api.get_all_orders(limit=20)
+        order_row = random.choice(orders['orders'])
+
+        order_no = order_row['orderNo']
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no,
+                                              field_type='text')
+
+        row = self.order_page.get_last_order_row()
+        self.orders_page.open_edit_page(row)
+
+        result = self.order_page.set_random_test_unit(test_units=random_row['name'], sub_order_index=0)
```

Comment:
self.order_page.set_random_test_unit(test_units=random_row['name'], sub_order_index=0) this function don't return anything in case test_units has a value !

Reply:
Done